cmd/av/commit_split.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"emperror.dev/errors"
+	"github.com/aviator-co/av/internal/git"
+	"github.com/aviator-co/av/internal/utils/colors"
+	"github.com/spf13/cobra"
+)
+
+var commitSplitCmd = &cobra.Command{
+	Use:          "split",
+	Short:        "split a commit into multiple commits",
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		repo, err := getRepo()
+		if err != nil {
+			return err
+		}
+		if clean, err := repo.CheckCleanWorkdir(); err != nil {
+			return err
+		} else if !clean {
+			return errors.New("the workdir is not clean")
```

Comment:
if this is a user facing message, you should explain it better
```suggestion
			return errors.New("The working directory is not clean, please stash or commit them before running split command")
```

Reply:
Also what's the difference between `errors.Errorf` and `errors.New`?

Reply:
See [`errors.Errorf`](https://pkg.go.dev/emperror.dev/errors#Errorf) and [`errors.New`](https://pkg.go.dev/emperror.dev/errors#New).

The tl;dr: is that `Errorf` allows using a format string while `New` only takes a single string argument.

Reply:
Done.